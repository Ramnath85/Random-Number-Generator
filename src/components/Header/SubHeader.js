import React from 'react';

const SubHeader = () => (
  <h2 className="App-subtitle">
    How many numbers do you want to generate?
  </h2>
);

export default SubHeader;