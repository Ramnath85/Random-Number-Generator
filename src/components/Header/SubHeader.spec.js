import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SubHeader from "./SubHeader";

configure({ adapter: new Adapter() });

describe('Test SubHeader component', () => {
  let shallowComponent;

  beforeEach(() => {
    shallowComponent = shallow(<SubHeader />);
  });

  test('should render once', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1)
  });
});