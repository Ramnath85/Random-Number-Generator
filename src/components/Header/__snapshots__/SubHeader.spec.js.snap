// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SubHeader component should render once 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "How many numbers do you want to generate?",
      "className": "App-subtitle",
    },
    "ref": null,
    "rendered": "How many numbers do you want to generate?",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "How many numbers do you want to generate?",
        "className": "App-subtitle",
      },
      "ref": null,
      "rendered": "How many numbers do you want to generate?",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
